src/index.ts

```diff
@@ -41,6 +46,12 @@ export class GraphQLClient {
       throw new ClientError({ ...errorResult, status: response.status}, {query, variables})
     }
   }
+
+  setHeader (headers: Headers): GraphQLClient {
```

Comment:
I suggest to rename this to `setHeaders` and add another `setHeader(key: string, value: string)` method.